### ISSUE TITLE
[FEATURE] Create Password Strength Validator

### ISSUE THREAD

**Issue by rubemfsv** [rubemfsv/clean-react-app#25]
**Description**

Ensures that a password meets certain strength criteria, such as minimum length, use of uppercase letters, numbers, and special characters.

**Expected Behavior**

The Password Strength Validator should evaluate the strength of a provided password based on predefined criteria, including:

1. Length Check: Verify that the password meets or exceeds a specified minimum length.
2. Character Types: Assess the presence of different character types, such as uppercase letters, lowercase letters, numbers, and special characters.
3. Complexity Rules: Apply additional rules or policies, such as not allowing commonly used passwords, dictionary words, or sequential patterns (e.g., "12345" or "password").
4. Strength Rating: Assign a strength rating to the password (e.g., weak, medium, strong) based on its compliance with the above criteria. (plus)
5. Provide Feedback: Offer clear feedback or suggestions on how the user can improve their password to meet stronger security standards. (plus)

This behavior ensures that the Password Strength Validator helps users create secure and robust passwords that meet the system's security requirements. It encourages the use of strong passwords, ultimately enhancing the security of user accounts and sensitive data.

**Additional Information**

Consider implementing this validator along with a corresponding unit test in the "validators" directory, alongside the existing validators.

**Comment by Aaketk17** [comment]
I will work on this please assign this to me. Thanks

**Comment by zainbinshakoor** [comment]
@rubemfsv Yes. Please assign this to me i will work on this

**Comment by rubemfsv** [comment]
@Aaketk17 done! Don't forget to ensure commit messages are following the conventional commits!
@zainbinshakoor sorry, but @Aaketk17 asked first. Please, check the other opened issues or suggest a new one 😄 if @Aaketk17 doesn’t do, I’ll sign to you. 
@Aaketk17 are you really going to do this one? I saw you got this and other task

**Comment by zainbinshakoor** [comment]
@rubemfsv sure why not

**Comment by Aaketk17** [comment]
@rubemfsv Thanks I will work on this you can assign the credit card validator task to @zainbinshakoor.

**Comment by rubemfsv** [comment]
@Aaketk17 are you still working on this?

**Comment by rubemfsv** [comment]
Issue unassigned due to the lack of answers. 

@zainbinshakoor if you want it, it's free now.

**Comment by zainbinshakoor** [comment]
yeah sure @rubemfsv i will solve it with in my free time

**Comment by zainbinshakoor** [comment]
where we need to implement this functionality?on signup? @rubemfsv

**Comment by rubemfsv** [comment]
@zainbinshakoor it would be nice there!

**Comment by zainbinshakoor** [comment]
@rubemfsv am currently busy hope we will work together in future

**Comment by SaurabPoudel** [comment]
Hello ! @rubemfsv , I am interested in contributing on this issue , as no one is assigned , could you please assign me ?

**Comment by vivekreddy049** [comment]
I would like to work on this

**Comment by rubemfsv** [comment]
@SaurabPoudel are you working on this?

**Comment by Nikhil-IITG** [comment]
@rubemfsv Please assign this issue to me

**Comment by rubemfsv** [comment]
@Nikhil-IITG are you still interested? done

**Comment by Nikhil-IITG** [comment]
> @Nikhil-IITG are you still interested? done

Yes, 'm quite stuck in the quizzes but will try to do it today or by tomorrow

**Comment by Nikhil-IITG** [comment]
Hello, can you please help me with the setup of .env file

**Comment by rubemfsv** [comment]
send me a message on linkedin https://www.linkedin.com/in/rubemfsv/ 
then I can help you!
